### PR TITLE
refactor: improve ranked choice voting readability and add test

### PR DESCRIPTION
**Pull request by wa0x6e**
This PR improves comprehension of the ranked choice voting system by adding more documentation, and add more test.

### Add more docs

Add more docs to some function, to help comprehension, and understand returned value.

### Add more tests

Add more tests, justifying that the fix https://github.com/snapshot-labs/snapshot.js/pull/1176 is really needed, and not just an edge case.

This fix is not related to invalid votes, etc..., the scenario can happen if the irv function pass on the first round, and the first round do no contain all the votes.

### Adjust `getChoiceString`  

Update function to handle invalid choice: they're now ignored, instead of showing `undefined`, like on the approval/quadratic/weighted voting system. 

This is consistent with the score computation, where those invalid votes are already ignored

### CODE REVIEW

**Comment by wa0x6e** [comment]
Not an important PR, don't need a release after merge, can wait next fix/feature

File: src/voting/rankedChoice.ts [hunk 1]
```diff
@@ -148,16 +187,57 @@ export default class RankedChoiceVoting {
...
+   * // Returns: "(1st) Alice" - invalid choice 5 is filtered out
+   */
   getChoiceString(): string {
     return this.selected
-      .map((choice) => {
-        if (this.proposal.choices[choice - 1])
-          return this.proposal.choices[choice - 1];
-      })
+      .map((choice) => this.proposal.choices[choice - 1])
+      .filter(Boolean)
```
Comment (ChaituVR) [inline src/voting/rankedChoice.ts:240]: Should be fine with reduced array length?

**Review by ChaituVR — APPROVED** [review]
tAck

File: src/voting/rankedChoice.ts (hunk 1 above)
Comment (ChaituVR) [reply src/voting/rankedChoice.ts:240]: We are filtering some choices here (before we are not filtering) so there will be difference in array length, i think it's fine but want to double check because PR is just refactor 😅

Comment (wa0x6e) [reply src/voting/rankedChoice.ts:240]: This is a little inconsistency I found while writing tests.

We are filtering out invalid choices everywhere, but single choice.

In which case this can break ? I don't think we check for arrays length anywhere before calling this function

Comment (ChaituVR) [reply src/voting/rankedChoice.ts:240]: Okies
